### PR TITLE
Add generic opt-in/out controller

### PR DESCRIPTION
**Pull request by desbo**
This way we don't have to keep copy/pasting these controllers. I think changing the routes might be an idea as well, but will need a bit of comms.

@johnduffell @OliverJAsh

### CODE REVIEW

**Comment by OliverJAsh** [comment]
Perfect :+1: 

Perhaps we could change the URLs to the new format, but keep the old ones around for the transition…

File: applications/conf/routes [hunk 1]
```diff
@@ -48,11 +48,11 @@ GET        /preferences
...
 
-GET        /https/optin                                                         controllers.HTTPSCookieController.optIn()
-GET        /https/optout                                                        controllers.HTTPSCookieController.optOut()
+GET        /https/optin                                                         controllers.OptInController.HTTPS.optIn()
+GET        /https/optout                                                        controllers.OptInController.HTTPS.optOut()
 
-GET        /new-header/optin                                                    controllers.NewHeaderCookieController.optIn()
-GET        /new-header/optout                                                   controllers.NewHeaderCookieController.optOut()
+GET        /new-header/optin                                                    controllers.OptInController.Header.optIn()
+GET        /new-header/optout                                                   controllers.OptInController.Header.optOut()
```
Comment (johnduffell) [inline applications/conf/routes]: if you restructure the urls I'll be even more happy! ;)

Comment (johnduffell) [inline applications/conf/routes]: however the existing https one would have to stay as it's too well known

**Comment by johnduffell** [comment]
:+1:

File: applications/conf/routes (hunk 1 above)
Comment (desbo) [inline applications/conf/routes]: i think olly's idea of using them both for a while will work

**Pull request merged**